atmo/jobs/models.py

```diff
@@ -100,12 +100,16 @@ def update_status(self):
         return self.most_recent_status
 
     def is_expired(self, at_time=None):
-        if self.current_run_jobflow_id is None:
-            return False  # job isn't even running at the moment
+        """Tells whether the current job run has run out of time or not"""
+        if not self.current_run_jobflow_id or not self.last_run_date:
+            # Job isn't even running at the moment
+            return False
         if at_time is None:
             at_time = timezone.now()
-        if self.last_run_date and self.last_run_date + timedelta(hours=self.job_timeout) >= at_time:
-            return True  # current job run expired
+        max_run_time = self.last_run_date + timedelta(hours=self.job_timeout)
+        if self.most_recent_status == "RUNNING" and at_time >= max_run_time:
```

Comment:
I think this also may be "WAITING"? Should we turn around the statement and use `if self.most_recent_status not in Cluster.FINAL_STATUS_LIST and at_time >= max_run_time`?